src/index.js

```diff
@@ -161,7 +161,12 @@ class UglifyJsPlugin {
                   'uglifyjs-webpack-plugin': versions.plugin,
                   'uglifyjs-webpack-plugin-options': this.options,
                   path: compiler.outputPath ? `${compiler.outputPath}/${file}` : file,
-                  hash: crypto.createHash('md5').update(input).digest('hex'),
+                  inputHash: crypto.createHash('md5').update(
+                    input ? input.toString() : '',
```

Comment:
Also other plugin can use own source implementation, we should manually call `toString()`